lalrpop/src/lr1/codegen/parse_table.rs

```diff
@@ -272,6 +275,28 @@ impl<'ascent, 'grammar, W: Write> CodeGenerator<'ascent, 'grammar, W, TableDrive
 
         rust!(self.out, "];");
 
+        // Only need the ERROR table if error recovery is used
+        if self.states.iter().any(|state| state.error.is_some()) {
+            // Error transitions which are positive integers for states that can be recovered from and zero
```

Comment:
Can we extend the big comment at the top that explains the overall strategy? In the particular, the part that starts "We create three parse tables:" :)